### PR TITLE
[Messaging] Sort folder messages

### PR DESCRIPTION
**Pull request by U-DON**
### Changes
- Fixed bug that was causing infinite fetching for the first page of a folder when no page query param was defined.
- Pulled SortableTable component into common (leaving original in Rx to remove in later PR).
- Implemented sorting for folder messages.
  - Works along with pagination as query parameters in URL.
  - Can directly access a page with a sort order.
- Pulled logic for disabling next and prev into MessageNav.
  - Resulted in a single function for changing pages in Folder.
- First pass at date formatting in folder messages table. Will fine-tune the behavior in separate PR.
### TODO
- Remove SortableTable from Rx and update its usage there.
- Pull out common table styling.
- Customize date formatting to our needs.

### CODE REVIEW

File: src/js/common/components/SortableTable.jsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        // Next click will be descending.
+        sortIcon = <i className="fa fa-caret-up"></i>;
+        nextSortOrder = 'DESC';
+      }
+    }
+
+    return (
+      <th key={field.value}>
+        <a onClick={this.handleSort(field.value, nextSortOrder)}>
+          {field.label}&nbsp;&nbsp;&nbsp;{sortIcon}
```
Comment (webinista) [inline src/js/common/components/SortableTable.jsx]: Using `&nbsp;&nbsp;&nbsp;`, which adds too much of an underline to the sort label. Add a `msg-sort` class to `sortIcon` and use CSS.

**Review by webinista — CHANGES_REQUESTED** [review]
One small CSS-y thing. LGTM otherwise.

File: src/js/common/components/SortableTable.jsx (hunk 1 above)
Comment (U-DON) [inline src/js/common/components/SortableTable.jsx]: Sure. Can I merge this and make that style change in the follow-up PR that moves the styles to a common class?

Comment (webinista) [inline src/js/common/components/SortableTable.jsx]: Yeah. That's fine.

**Pull request merged**